Source/ACE/Entity/Landblock.cs

```diff
@@ -543,19 +455,36 @@ public void UseTime()
                         }
                     });
 
-                    if (mo.Location.LandblockId == id)
-                    {
-                        // update if it's still here
-                        Broadcast(BroadcastEventArgs.CreateAction(BroadcastAction.AddOrUpdate, mo), true, Quadrant.All);
-                    }
-                    else
+                    if (mo.Location.LandblockId != id)
                     {
                         // remove and readd if it's not
                         RemoveWorldObject(mo.Guid, false);
                         LandblockManager.AddObject(mo);
                     }
                 });
 
+                // Tick all objects
+                double tickTime = WorldManager.PortalYearTicks;
+                // per-creature update on landblock.
+                Parallel.ForEach(allworldobj, wo =>
+                {
+                    // Process the creatures
+                    // We lock here -- it IS required becuase this is run in a Parallel.ForEach loop
+                    //   AND because of cross-landblock dependencies
+                    //   Any use of OTHER objects within the tick function (gotten through landblock by guid) requires locking
+                    //   updates of THIS object within Tick are safe
+                    var tickable = wo as ITickable;
+                    if (tickable != null)
+                    {
+                        lock (tickable)
+                        {
+                            // FIXME(ddevec): LBBroadcaster class was a fast hacky fix to the problem -- there is probably a more elegant solution..
```

Comment:
I had the impression that the locker object inside the loop would be declared/instantiated per thread.  Therefore, locking it would be like locking any other non-shared resource: a no-op.  I could very well be wrong, but that's always been my impression of how .Net executes lambdas.